artiq/applets/plot_xy.py

```diff
@@ -24,32 +31,57 @@ def data_changed(self, data, mods, title):
         fit = data.get(self.args.fit, (False, None))[1]
 
         if not len(y) or len(y) != len(x):
-            return
+            self.mismatch['X values'] = True
+            self.timer.start(1000)
+        else:
+            self.mismatch['X values'] = False
+            if not sum(self.mismatch.values()):
+                self.timer.stop()
         if error is not None and hasattr(error, "__len__"):
             if not len(error):
                 error = None
             elif len(error) != len(y):
-                return
+                self.mismatch['Error bars'] = True
+                self.timer.start(1000)
+            else:
+                self.mismatch['Error bars'] = False
+                if not sum(self.mismatch.values()):
+                    self.timer.stop()
         if fit is not None:
             if not len(fit):
                 fit = None
             elif len(fit) != len(y):
-                return
+                self.mismatch['Fit values'] = True
+                self.timer.start(1000)
+            else:
+                self.mismatch['Fit values'] = False
+                if not sum(self.mismatch.values()):
+                    self.timer.stop()
 
         self.clear()
+        if self.mismatch['X values']:
+            return
```

Comment:
This deviates from the original behavior AFAICT. Why?

Reply:
If the length of X and Y does not match, `self.plot` at the next line will cause the applet to crash and has to be reopened.

Reply:
That would be a separate bug, can you fix it first in a separate PR?

Reply:
My fault, the previous explanation is incorrect. In the original file, it will not show anything if the `error bars` or `fit values` do not have the same length as `Y`. But I would like to change it. I want to plot whatever is possible and show the error message below. Should I do that or stick with the original behavior.

Reply:
Probably easier to keep the original behavior.